### ISSUE TITLE
slf4j-api version change from 1.7.x to 2.0.x between morphia 2.2.8 and 2.2.9 is a breaking change

### ISSUE THREAD

**Issue by grossws** [MorphiaOrg/morphia#2124]
**Describe the bug**
Morphia 2.2.9 updates `slf4j-api` from `1.7.x` to `2.0.x` which breaks downstream consumers that have slf4j implementation targeting 1.7.x.

**To Reproduce**

1. Create simple project with `dev.morphia.morphia:morphia-core:2.2.9` and `ch.qos.logback:logback-classic:1.2.11` in dependencies and try log anything.

```kotlin
// settings.gradle.kts
rootProject.name = "morphia-slf4j"
dependencyResolutionManagement {
  repositories.mavenCentral()
}
```

```kotlin
// build.gradle.kts
plugins { java; application }
application { mainClass.set("Main") }
dependencies {
  implementation("dev.morphia.morphia:morphia-core:2.2.9")
  implementation("ch.qos.logback:logback-classic:1.2.11")
}
```

```java
// src/main/java/Main.java
public class Main {
    public static void main(String[] args) {
        org.slf4j.LoggerFactory.getLogger(Main.class).info("ok");
    }
}
```

2. Run `gradle run`
3. You get following message:
```
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#noProviders for further details.
SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions prior to 1.8.
SLF4J: Ignoring binding found at [jar:file:/home/gross/.gradle/caches/modules-2/files-2.1/ch.qos.logback/logback-classic/1.2.11/4741689214e9d1e8408b206506cbe76d1c6a7d60/logback-classic-1.2.11.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#ignoredBindings for an explanation.
```

**Expected behavior**

Breaking change should at least be documented and avoided at the very least in patch release.

** Please complete the following information: **
 - Server Version: N/A
 - Driver Version: N/A
 - Morphia Version: 2.2.9

**Comment by evanchooly** [comment]
Yep, you're absolutely right.  I got a little aggressive trying to keep 2.2 and master in sync.  Now the question, do I roll it back and potentially rebreak folks that have upgraded?  Or just leave it as is and count it as a lesson learned and something to be more mindful about in the future.  I'm inclined to leave it.

Either way, I apologize.  In trying to wrap up 3 different releases, I got a little carried away.  I'll do better in the future.

**Comment by grossws** [comment]
In my opinion it's better to rollback and update on major release or in at least two minor releases but it's up to you ,)

There's a workaround for the most popular logging backends except `slf4j-jboss-logmanager` (used in [Quarkus](https://quarkus.io) for example). Also downgrade may be viable solution for downstream users if none of the new APIs in the `slf4j` are called (fluent logging API for example). 

For more info and available logging libraries updated see the thread in [Apache Tika dev mailing list](https://lists.apache.org/thread/7mor46vn5xk1jp1t8lo1vczdcrqtktcb).

**Comment by grossws** [comment]
I just downgraded Morphia to 2.2.8 in the project with Quarkus (and thus `slf4j-jboss-logmanger`)

**Comment by evanchooly** [comment]
I'll roll it back.  I'm about to cut a 2.2.10 release anyway which will, fingers crossed, be (one of?) the last releases of 2.2.x since 2.3 will soon be out.

If you're using quarkus, have you seen https://github.com/quarkiverse/quarkus-morphia?  There's not a *lot* there yet but it works with quarkus and might be useful-ish.  It will be better once 2.3 lands (and thus critter), though.

**Comment by grossws** [comment]
Haven't tried it yet. I'm migrating legacy project that recently had `org.mongodb.morphia:morphia:0.111` and `org.mongodb:mongo-java-driver:2.14.x` with multiple datastores and mongodb connections (2-4 per service instance at least) so simple solutions like just using `@Inject MongoClient` from `quarkus-mongodb-client` doesn't work in my case unfortunately. From a brief glance such case is out of scope for your extension as of moment.

I'll try it on some less convoluted project to evaluate it and share feedback. Our case is likely quite niche since primary feature that we use Morphia for is optimistic locks with `@Version`.

For perspective the project I'm talking about was actively developed since 2009-2010, run on GlassFish, then JBoss AS and then on our home grown server (on top of embedded Jetty, RestEasy, Weld, Jasig CAS/Atlassian Crowd and bunch of other libraries from early 2010s), was migrated from pure J2EE to a few ubiquitous Java EE specs like JAX-RS/CDI/Servlet API over time, has it's own reflection-based plugin infrastructure for CDI/JAX-RS from the times of CDI 1.2 long before Swagger/OpenAPI became popular et cetera, et cetera.

**Comment by evanchooly** [comment]
Wow.  A classic!  I'm happy to answer any questions you run in to.  And if you try quarkus-morphia and find a gap, I can take a look.  I haven't touched in a while because it's not a *huge* priority at work and I'm unsure of what would be the best use of time to evolve it.

**Comment by grossws** [comment]
Ah, thanks, I'll keep that in mind. Are you on Quarkus' Zulip server? Or better to just use discussions here?

**Comment by evanchooly** [comment]
Yep.  I'm on zulip.  if your questions are purely morphia, though, perhaps use the discussions tab here.

**Comment by evanchooly** [comment]
I just pushed 2.2.10.  It should be ready in maven central in an hour or two.

**Comment by grossws** [comment]
Thanks for update ,)
